pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/MedicalRecordFinalObjectTransformer.java

```diff
@@ -0,0 +1,51 @@
+package org.broadinstitute.dsm.model.elastic;
+
+import java.util.Map;
+import java.util.Objects;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import org.broadinstitute.dsm.db.MedicalRecordFinalDto;
+import org.broadinstitute.dsm.model.elastic.converters.camelcase.CamelCaseConverter;
+import org.broadinstitute.dsm.model.elastic.export.parse.ValueParser;
+import org.broadinstitute.dsm.model.elastic.export.parse.abstraction.MedicalRecordAbstractionFieldType;
+import org.broadinstitute.dsm.model.elastic.export.parse.abstraction.source.MedicalRecordAbstractionSourceGenerator;
+import org.broadinstitute.dsm.model.elastic.export.parse.abstraction.source.MedicalRecordAbstractionSourceGeneratorFactory;
+import org.broadinstitute.dsm.model.elastic.migration.MedicalRecordFinalColumnNameBuilder;
+import org.broadinstitute.dsm.model.elastic.migration.MedicalRecordFinalColumnNameBuilderLive;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.util.proxy.jackson.ObjectMapperSingleton;
+
+
+/**
+ * The class which is responsible for transforming `medical_record_final` objects into the map instances suited for ElasticSearch.
+ */
+public class MedicalRecordFinalObjectTransformer extends ObjectTransformer {
+
+    MedicalRecordFinalColumnNameBuilder columnBuilder;
+
+    public MedicalRecordFinalObjectTransformer(String realm) {
+        super(realm, new ValueParser());
+        this.columnBuilder = new MedicalRecordFinalColumnNameBuilderLive(CamelCaseConverter.of());
+    }
+
+    /**
+     * Transforms each object to the map representation
+     * @param obj a concrete object of `ddp_medical_record_final`
+     */
+    @Override
+    public Map<String, Object> transformObjectToMap(Object obj) {
+        MedicalRecordFinalDto medicalRecordFinalDto = (MedicalRecordFinalDto) obj;
+        MedicalRecordAbstractionFieldType fieldType = MedicalRecordAbstractionFieldType.of(medicalRecordFinalDto.getType());
+        MedicalRecordAbstractionSourceGenerator sourceGenerator = MedicalRecordAbstractionSourceGeneratorFactory.spawn(fieldType);
+        Map<String, Object> result = ObjectMapperSingleton.readValue(
+                ObjectMapperSingleton.writeValueAsString(medicalRecordFinalDto),
+                new TypeReference<Map<String, Object>>() {});
```

Comment:
I think it is better to use `ObjectMapperSingleton.instance().convertValue` here instead of serialize and deserialize as string.